crates/gpui/src/platform/linux/wayland/window.rs

```diff
@@ -325,10 +327,14 @@ impl WaylandWindowStatePtr {
                 let height = NonZeroU32::new(height as u32);
                 let fullscreen = states.contains(&(xdg_toplevel::State::Fullscreen as u8));
                 let maximized = states.contains(&(xdg_toplevel::State::Maximized as u8));
-                self.resize(width, height);
-                self.set_fullscreen(fullscreen);
                 let mut state = self.state.borrow_mut();
                 state.maximized = maximized;
+                state.fullscreen = fullscreen;
+                if fullscreen || maximized {
+                    state.restore_bounds = state.bounds.map(|p| DevicePixels(p as i32));
+                }
+                self.resize(width, height);
```

Comment:
Should call `drop(state)` before calling `self.resize(...)` to prevent a double borrow.

Reply:
Thank you. BTW, the `WaylandDisplay` is marked `todo`, so there maybe another PR to address this in the future. For now, I guess the main display is returned.